examples/pigweed-app/esp32/include/AppConfig.h

```diff
@@ -0,0 +1,54 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+// ESP32 Logging
+// ---- PW Example App Config ----
+
+#include "PigweedLoggerMutex.h"
+#include <support/logging/CHIPLogging.h>
+/*
+ * General-purpose logging function
+ */
+
+extern "C" void esp32Log(const char *log, const char * TAG, const char * aFormat, ...)
+{
+    va_list v;
+    va_start(v, aFormat);
+#if CONFIG_LOG_DEFAULT_LEVEL > 1
+    char formattedMsg[CHIP_CONFIG_LOG_MESSAGE_MAX_SIZE];
+    strcpy(formattedMsg, TAG);
+    strcat(formattedMsg, ": [");
+    strcat(formattedMsg, log);
+    strcat(formattedMsg, "] ");
+    size_t prefixLen = strlen(formattedMsg);
+    size_t len       = vsnprintf(formattedMsg + prefixLen, sizeof formattedMsg - prefixLen, aFormat, v); 
+    if (len >= sizeof formattedMsg - prefixLen)
+    {   
+        formattedMsg[sizeof formattedMsg - 1] = '\0';
+    }   
+    PigweedLogger::putString(formattedMsg, len+prefixLen);
+#endif 
+    va_end(v);
+}
+
+#define ESP_LOGE(...) esp32Log("ERROR", __VA_ARGS__);
```

Comment:
I think, for this example, all the ESP_LOG* messages from ESP-IDF, CHIP and this example should be mapped/changed as per your requirement. The current change will only modify the ESP_LOG* messages used in the example.

Reply:
Changes added using wrapup function for `esp_log_write`.